### DISCUSSION TITLE
1.9a2

### DISCUSSION THREAD

**Discussion by sgillies** [Toblerity/Fiona#1117] [Announcements]
The second 1.9 pre-release is ready for early testing. Not all of the features planned for 1.9.0 are in this release, and some of the features here may yet be modified. Please pay close attention to the deprecations and packaging changes.

The binary wheels on PyPI include GDAL 3.5.0 and PROJ 9.0.0, the latest versions of each.

Deprecations:

- Fiona's API methods will accept feature and geometry dicts in 1.9.0, but this usage is deprecated. Instances of Feature and Geometry will be required in 2.0.
- The precision keyword argument of fiona.transform.transform_geom is deprecated and will be removed in version 2.0.
- Deprecated usage has been eliminated in the project. Fiona's tests pass when run with a -Werror::DeprecationWarning filter.

Changes:

- Fiona's FionaDeprecationWarning now sub-classes DeprecationWarning.
- Some test modules have beeen re-formatted using black.

New features:

- Fiona Collections now carry a context exit stack into which we can push fiona Envs and MemoryFiles (#1059).
- Fiona has a new CRS class, like rasterio's, which is compatible with the CRS dicts of previous versions (#714).

<hr /><em>This discussion was created from the release <a href='https://github.com/Toblerity/Fiona/releases/tag/1.9a2'>1.9a2</a>.</em>

**Comment by sgillies** [comment]
Wheels for 64-bit Windows systems have been added to https://pypi.org/project/Fiona/1.9a2/#files. Would love to hear if they work for you!

**Reply by elfjes** [reply]
This is great! I love the effort to provide binary windows wheels for Fiona. It's very important for us to provide a smooth installation experience for our users, and being forced to either use anaconda (fantastic as that environment may be) or get wheels from some third party location (https://www.lfd.uci.edu/~gohlke/pythonlibs/#fiona) is a less than ideal situation. There is huge value in just being able to `pip install` a package with all it's dependencies.

**Reply by devmcp** [reply]
Fantastic work. I've been using these for a while now with no issues. Many thanks!
